### PR TITLE
Log type metrics

### PR DESCRIPTION
**Pull request by mlw**

### CODE REVIEW

File: Source/santad/SNTApplicationCoreMetrics.mm [hunk 1]
```diff
@@ -56,7 +79,7 @@ static void RegisterMemoryAndCPUMetrics(SNTMetricSet *metricSet) {
                        fieldNames:@[]
                          helpText:@"The virtual memory size of this process"];
   SNTMetricInt64Gauge *rsize =
-    [metricSet int64GaugeWithName:@"/proc/memory/resident_size"
+    [metricSet int64GaugeWithName:@ "/proc/memory/resident_size"
```
Comment (russellhancox) [inline Source/santad/SNTApplicationCoreMetrics.mm:82]: Remove space after @. Surprised the linter didn't catch this.

File: Source/santad/SNTApplicationCoreMetrics.mm [hunk 2]
```diff
@@ -47,6 +47,29 @@ static void RegisterModeMetric(SNTMetricSet *metricSet) {
   }];
 }
 
+/**
+ * Register the event log type metric checking the config before reporting the status.
+ */
+static void RegisterModeEventLogType(SNTMetricSet *metricSet) {
```
Comment (russellhancox) [inline Source/santad/SNTApplicationCoreMetrics.mm:53]: RegisterEventLogType

File: Source/santad/SNTApplicationCoreMetrics.mm (hunk 1 above)
Comment (pmarkowsky) [reply Source/santad/SNTApplicationCoreMetrics.mm:82]:
```suggestion
    [metricSet int64GaugeWithName:@"/proc/memory/resident_size"
```

**Review by pmarkowsky — APPROVED** [review]
LGTM.

**Pull request merged**